pkg/registry/apis/provisioning/resources/id.go

```diff
@@ -92,12 +92,12 @@ func appendHashSuffix(hashKey, repositoryName string) func(string) string {
 }
 
 // Will pick a name based on the hashed repository and path
-func NamesFromHashedRepoPath(repo string, fpath string) (string, string) {
+func FileNameFromHashedRepoPath(repo string, fpath string) string {
 	// Remove the extension: we don't want the extension to impact the ID. This lets the user change between all supported formats.
 	fpath = safepath.RemoveExt(fpath)
 	hasher := appendHashSuffix(fpath, repo)
 
-	return hasher(safepath.Base(fpath)), hasher(strings.Trim(safepath.Dir(fpath), "/"))
+	return hasher(safepath.Base(fpath))
```

Comment:
This was the bug for the directory name. For sync the name will be the dir name and when using `/files` api a trimmed version of entire dir path. 

I opted for simple the dir name only for clarity and scalability. 


![image](https://github.com/user-attachments/assets/dc8b0222-b9ef-4479-a4fb-957490cac953)

Reply:
I think we can even shorten the hash part -- it is kinda ugly :)

Reply:
sure, I will shorten it when I add more tests in a follow-up.